### PR TITLE
python310Packages.twisted: 22.8.0 -> 22.10.0

### PR DESCRIPTION
**Pull request by dotlambda**
What do we do about 22.05?

###### Description of changes
fixes CVE-2022-39348
https://github.com/twisted/twisted/releases/tag/twisted-22.10.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @NixOS/nixos-release-managers

### CODE REVIEW

**Comment by mweinelt** [comment]
> What do we do about 22.05?

Mangle the patches mentioned in the advisory.

File: pkgs/development/python-modules/twisted/default.nix [hunk 1]
```diff
@@ -128,6 +121,7 @@ buildPythonPackage rec {
   checkInputs = [
     git
     glibcLocales
+    hypothesis
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/twisted/default.nix:124]: That explains why some tests failed when I tried to update and couldn't find an error log.

**Comment by vcunat** [comment]
So, this broke the build/tests on aarch64-darwin:
 - https://hydra.nixos.org/build/200269778
 - ofBorg attempt in 22.11 backport
   https://logs.nix.ci/?key=nixos/nixpkgs.203401&attempt_id=a4a5e395-6893-4128-a822-b8c5e87c5458

(The other three platforms built OK on `staging-next`.)

**Comment by vcunat** [comment]
This update broke `buildbot`.  https://hydra.nixos.org/build/200806424
/cc maintainers: @ryansydnor, @lopsided98.
Also in 22.11; it's not nice when security fixes bring incompatible changes (e.g. "deprecations and removals" in these release notes).

**Pull request merged**